### ISSUE TITLE
Lack of dynamic library libnghttp2.so.14 when starting PSME service

### ISSUE THREAD

**Issue by nicwu-cel** [opencomputeproject/HWMgmt-DeviceMgr-PSME#3]
Hi Developers,

When I tried to start PSME service using "service psme start". It didn't work as below.
root@localhost:~#  service psme start
start_daemon
start_daemon
root@localhost:~# service psme status
psme-rest-server is not running.
root@localhost:~#

After I using 
root@localhost:~# /usr/local/bin/psme-rest-server

It happens:

/usr/local/bin/psme-rest-server: error while loading shared libraries: libnghttp2.so.14: cannot open shared object file: No such file or directory

I wonder if this dynamic library libnghttp2.so.14 should be included in ONL OS or PSME DEB package?

**Comment by jcleung5549** [comment]
Would you mind trying the PSME contribution at - https://github.com/opencomputeproject/Rack-Manager/tree/master/Contrib-Intel/RSD-PSME-RMM?  I'd be interested you see the problem?

**Comment by nicwu-cel** [comment]
May I know what's the relationship between 
https://github.com/opencomputeproject/HWMgmt-DeviceMgr-PSME 
and 
https://github.com/opencomputeproject/Rack-Manager/tree/master/Contrib-Intel/RSD-PSME-RMM? 

Are they the same method to build redfish service package?

**Comment by jcleung5549** [comment]
The RSD-PSME-RMM contributed from Intel RackScale Design effort (circa 2016).  RSD had three Redfish interfaces with different management domains, each running as a in-band agent.  PSME's management domain was the pod.  The repository was contribute to OCP, at the start of the OpenRMC subproject.  The Redfish models were evolving quickly during 2016, models implemented are dated.  But everything built when it was contributed.

The DM-PSME was contributed as part of the Device Manager contribution (2021),  The DM-PSME was derived from the RSD-PSME repository.  DM-PSME was modified to conform to the OCP Baseline profile and execute on ONIE/ONL.
See slide 5 in the following presentation - https://drive.google.com/file/d/1ZLk9IjGRsFCsC-smoqDY9ohiCxBavEg9

The subproject has focused on the Device Manager portion of the contribution and haven't touched the DM-PSME contribution.

I asked the original PSD-PSME developers for their take on this issue.  Their response was "Regarding this issue, our implementation used libmicrohttpd for HTTP. I have not even heard of libnghttp2 before today. This must be related to the changes."

**Comment by nicwu-cel** [comment]
Thanks for the detailed response and sorry for the late response.

I found some differences when I built PSME in different compile servers.

- On compile server 1:

```
~# ldd ./PSME/build/install/allinone-deb/psme-allinone/usr/local/bin/psme-rest-server
...
libnghttp2.so.14 => /lib/x86_64-linux-gnu/libnghttp2.so.14 (0x00007f4e08145000)
...
~#  ls /lib/x86_64-linux-gnu/libnghttp2.so.14
/lib/x86_64-linux-gnu/libnghttp2.so.14

```
- On compile server 2, it no more required ibnghttp2.so.14 but libidn.so.11:
```
~# ldd ./PSME/build/install/allinone-deb/psme-allinone/usr/local/bin/psme-rest-server
...
libidn.so.11 => /usr/lib/x86_64-linux-gnu/libidn.so.11 (0x00007f0d21101000)
...
~# ls  /usr/lib/x86_64-linux-gnu/libidn.so.11
/usr/lib/x86_64-linux-gnu/libidn.so.11
```

Actually , bin file "psme-rest-server" only relies on libmicrohttpd.so.12
```
root@localhost:~# readelf -a /usr/local/bin/psme-rest-server | grep http
 0x0000000000000001 (NEEDED)             Shared library: [libmicrohttpd.so.12]
root@localhost:~#
```

I wonder if this issue is caused by the host environment building PSME which incorrectly lead some extra library dependencies to "psme-rest-server"

**Comment by jcleung5549** [comment]
What version of Linux's are you compiling on? And are you running the ./pre-inst-X86-pkgs.sh?

**Comment by nicwu-cel** [comment]
What version of Linux's are you compiling on?
R: 
Compile server 1:
Host and docker version are both 5.4.0-125-generic
Compile server 2:
Host and docker version are both 4.15.0-142-generic

And are you running the ./pre-inst-X86-pkgs.sh?
R: 
yes, I've ran ./pre-inst-X86-pkgs.sh before ./Build_all.sh



When I attempted to delete libnghttp2.so.14 before ./Build_all.sh after  ./pre-inst-X86-pkgs.sh, I got below error message
```
[ 42%] Linking CXX executable ../../../../bin/tests/unittest_agent-framework_configuration
/usr/bin/ld: warning: libnghttp2.so.14, needed by /usr/lib/gcc/x86_64-linux-gnu/8/../../../x86_64-linux-gnu/libcurl.so, not found (try using -rpath or -rpath-link)
```

libnghttp2.so.14 is needed by /usr/lib/x86_64-linux-gnu/libcurl.so

PSME reuses /usr/lib/x86_64-linux-gnu/libcurl.so and copy it to PSME DEB package in prepare.sh as below

`PSME/build/install/allinone-deb/sh/prepare.sh:    cp $LIB_PREINSTALL/libcurl.so.4                       $ITEM_PATH/usr/local/lib
`
If /usr/lib/x86_64-linux-gnu/libcurl.so relies on libnghttp2.so.14 in docker but prepare.sh doesn't copy it from docker as libcurl to PSME DEB package, the lack of libnghttp2 will happen.

So I checked the dependencies of libnghttp2 and it did need libnghttp2:

```
~# ldd /usr/lib/x86_64-linux-gnu/libcurl.so | grep http
        libnghttp2.so.14 => /usr/lib/x86_64-linux-gnu/libnghttp2.so.14 (0x00007fd2f045b000)
```

The version of libcurl.so in docker is 4.5.0 because it links to libcurl-nss.so.4.5.0

Exiting the docker and checking libcurl.so under the host and found libcurl.so didn't relied on libnghttp2, it links to libcurl.so.4.4.0.

So I think this issue is caused by libcurl update.
